trl/trainer/grpo_trainer.py

```diff
@@ -815,12 +833,37 @@ def _get_per_token_logps(self, model, input_ids, attention_mask, logits_to_keep,
             # For transformers<=4.48, logits_to_keep argument isn't supported, so here we drop logits ourselves.
             # See https://github.com/huggingface/trl/issues/2770
             logits = logits[:, -logits_to_keep:]
-            # Divide logits by sampling temperature.
-            # See https://huggingface.co/blog/the_n_implementation_details_of_rlhf_with_ppo#policy-training-implementation-details
-            logits = logits / self.temperature
-            logps = selective_log_softmax(logits, input_ids_batch)  # compute logprobs for the input tokens
+
+            # Divide logits by sampling temperature
+            scaled_logits = logits / self.temperature
+
+            # Compute full probability distributions
+            probs = torch.softmax(scaled_logits, dim=-1)  # (B, seq_len, vocab_size)
```

Comment:
It seems to me that this would introduce a big memory spike. In fact, this was the reason for using `selective_log_softmax` in the first place, as the memory spike actually occurs here.
What's more, all results (`probs`) are kept in VRAM.

Reply:
Thanks, let me try `selective_log_softmax` and clean up the VRAM